src/Azure/Orleans.Persistence.AzureStorage/Providers/Storage/AzureTableStorage.cs

```diff
@@ -498,7 +498,8 @@ public static class AzureTableGrainStorageFactory
         public static IGrainStorage Create(IServiceProvider services, string name)
         {
             IOptionsSnapshot<AzureTableStorageOptions> optionsSnapshot = services.GetRequiredService<IOptionsSnapshot<AzureTableStorageOptions>>();
-            return ActivatorUtilities.CreateInstance<AzureTableGrainStorage>(services, optionsSnapshot.Get(name), name);
+            IClusterSettings clusterSettings = services.GetServiceByNameOrDefault<IClusterSettings>(name);
```

Comment:
If the application layer registers a populated ClusterOptions by name as an IClusterSettings service, it will be picked up here.  If they don't the clusteroptions defined for the service will be used.  Any provider that does not do this can access the cluster options via IOptions, as is currently done, or via the default IClusterSettings interface.